lib/ztp_helper.py

```diff
@@ -132,15 +132,28 @@ def setup_debug_logger(self):
 
         self.logger = logger
 
+    def read_in_chunks(self, file_object, chunk_size=1048576):
+        """generator to read the file in 1MB chunks
+           :param file_object: File to be read
+           :param chunk_size: Chunk size to read in every iteration
+           :type file_object: int
+           :type chunk_size: int
+        """
+        while True:
+            data = file_object.read(chunk_size)
+            if not data:
+                break
+            yield data
 
-
-    def download_file(self, file_url, destination_folder):
+    def download_file(self, file_url, destination_folder, md5sum=None):
```

Comment:
@mvshpavan  Pass the parameter `chunk_size` to the download_file function so that it may in turn pass it into the read_in_chunks() method.  With the current code, the chunk_size is hardcoded.